hbase-server/src/main/java/org/apache/hadoop/hbase/wal/WALPrettyPrinter.java

```diff
@@ -301,15 +318,12 @@ public void processFile(final Configuration conf, final Path p)
         List<Map<String, Object>> actions = new ArrayList<>();
         for (Cell cell : edit.getCells()) {
           // add atomic operation to txn
-          Map<String, Object> op = new HashMap<>(toStringMap(cell, outputOnlyRowKey));
-          if (outputValues) {
-            op.put("value", Bytes.toStringBinary(CellUtil.cloneValue(cell)));
-          }
-          // check row output filter
-          if (row == null || ((String) op.get("row")).equals(row)) {
```

Comment:
For me I think this is only place we need to change? Just filtering with rowPrefix? Seems you have done other refactorings? What is the purpose?

Reply:
@Apache9 Yes, initially I thought the same way and did that. But @wchevreuil brings the valid concern that rowprefix filter and row filter should both be available.
Example from @wchevreuil :
If there are rows `a`, `aa`, `aaa` and user just wants row `a`, the rowPrefix will not help in this case. Hope it answers your concern.

Reply:
My concern is that, we just need to add an extra 'if' here to test for rowPrefix? Why we completely move the test for row and rowPrefix to other place?

Reply:
Agree that moving the `if` check from this `processFile` method all the way to `toStringMap` may not be needed, but didn't think it's much of a concern, so kept my +1. Do you see this as a blocker, @Apache9 ?

Reply:
I do not mean that we must change the code to the old style. Just want to know the reason why we do not use the simple way.

Reply:
I see what you mean, I changed the flow because it was not very readable even through it was functionally correct. Here is why I thought so:
older version
```java
          Map<String, Object> op = new HashMap<>(toStringMap(cell, outputOnlyRowKey));
          if (outputValues) {
            op.put("value", Bytes.toStringBinary(CellUtil.cloneValue(cell)));
          }

          // check row output filter
          if (row == null || ((String) op.get("row")).equals(row)) {
            actions.add(op);
          }
          op.put("total_size_sum", cell.heapSize());
```

Here, we are computing the stringMap and then checking if we need this row, and then even if don't need the row we still compute `total_size_sum`. Not sure why it was done this way but this was my concern. Hope it answers your question.

Reply:
@Apache9

Reply:
OK, got it. No problem.